pkg/sql/column_type.go

```diff
@@ -26,104 +26,8 @@ import (
 // hive column type ends with _TYPE
 const hiveCTypeSuffix = "_TYPE"
 
-var (
-	// Column types: https://golang.org/pkg/database/sql/#Rows.Scan
-	sqlNullBool    = reflect.TypeOf(sql.NullBool{})
-	sqlNullInt64   = reflect.TypeOf(sql.NullInt64{})
-	sqlNullFloat64 = reflect.TypeOf(sql.NullFloat64{})
-	sqlRawBytes    = reflect.TypeOf(sql.RawBytes{})
-	sqlNullString  = reflect.TypeOf(sql.NullString{})
-	mysqlNullTime  = reflect.TypeOf(mysql.NullTime{})
-	// builtin type supports sql like `select 1;` or  `select count(*) from ...`
-	builtInBytes   = reflect.TypeOf([]byte(""))
-	builtinString  = reflect.TypeOf(string(""))
-	builtinInt     = reflect.TypeOf(int(0))
-	builtinInt8    = reflect.TypeOf(int8(0))
-	builtinInt16   = reflect.TypeOf(int16(0))
-	builtinInt32   = reflect.TypeOf(int32(0))
-	builtinInt64   = reflect.TypeOf(int64(0))
-	builtinUint    = reflect.TypeOf(uint(0))
-	builtinUint8   = reflect.TypeOf(uint8(0))
-	builtinUint16  = reflect.TypeOf(uint16(0))
-	builtinUint32  = reflect.TypeOf(uint32(0))
-	builtinUint64  = reflect.TypeOf(uint64(0))
-	builtinFloat32 = reflect.TypeOf(float32(0))
-	builtinFloat64 = reflect.TypeOf(float64(0))
-	builtinTime    = reflect.TypeOf(time.Time{})
-
-	instantiators = map[reflect.Type]func() interface{}{
-		sqlNullBool: func() interface{} {
-			return new(sql.NullBool)
-		},
-		sqlNullInt64: func() interface{} {
-			return new(sql.NullInt64)
-		},
-		sqlNullFloat64: func() interface{} {
-			return new(sql.NullFloat64)
-		},
-		sqlRawBytes: func() interface{} {
-			return new(sql.RawBytes)
-		},
-		sqlNullString: func() interface{} {
-			return new(sql.NullString)
-		},
-		mysqlNullTime: func() interface{} {
-			return new(mysql.NullTime)
-		},
-		builtinTime: func() interface{} {
-			return new(time.Time)
-		},
-		builtInBytes: func() interface{} {
-			return new([]byte)
-		},
-		builtinString: func() interface{} {
-			return new(string)
-		},
-		builtinInt: func() interface{} {
-			return new(int)
-		},
-		builtinInt8: func() interface{} {
-			return new(int8)
-		},
-		builtinInt16: func() interface{} {
-			return new(int16)
-		},
-		builtinInt32: func() interface{} {
-			return new(int32)
-		},
-		builtinInt64: func() interface{} {
-			return new(int64)
-		},
-		builtinUint: func() interface{} {
-			return new(uint)
-		},
-		builtinUint8: func() interface{} {
-			return new(uint8)
-		},
-		builtinUint16: func() interface{} {
-			return new(uint16)
-		},
-		builtinUint32: func() interface{} {
-			return new(uint32)
-		},
-		builtinUint64: func() interface{} {
-			return new(uint64)
-		},
-		builtinFloat32: func() interface{} {
-			return new(float32)
-		},
-		builtinFloat64: func() interface{} {
-			return new(float64)
-		},
-	} // end of instantiators.
-)
-
-func createByType(rt reflect.Type) (interface{}, error) {
-	f, ok := instantiators[rt]
-	if ok {
-		return f(), nil
-	}
-	return nil, fmt.Errorf("unrecognized column scan type %v", rt)
+func createByType(t reflect.Type) interface{} {
```

Comment:
The meaning of `createByType` is not clear.

Reply:
What is your suggestion?

Reply:
3x for pointing out this, I'll come up with a better name later.